### PR TITLE
Guard clause

### PR DESCRIPTION
**Pull request by evanrolfe**
https://trello.com/c/ZXHXwSOM/178-enable-the-style-guardclause-cop-in-rubocop

### CODE REVIEW

File: src/api/app/controllers/public_controller.rb [hunk 1]
```diff
@@ -186,11 +186,8 @@ def binary_packages
 
   # removes /private prefix from path
   def unshift_public(path)
-    if path =~ %r{/public(.*)}
-      return $1
-    else
-      return path
-    end
+    return $1 if path =~ %r{/public(.*)}
+    return path
```
Comment (Ana06) [inline src/api/app/controllers/public_controller.rb]: Lines 189 & 190 can be written as:
``` Ruby
path =~ %r{/public(.*)} ? $1 : path

File: src/api/app/controllers/webui/package_controller.rb [hunk 2]
```diff
@@ -127,10 +127,10 @@ def dependency
     @fileinfo = Fileinfo.find(project: params[:dproject], package: '_repository', repository: params[:drepository], arch: @arch,
       filename: params[:dname], view: 'fileinfo_ext')
     @durl = nil
-    unless @fileinfo # avoid displaying an error for non-existing packages
-      redirect_back(fallback_location: { action: :binary, project: params[:project], package: params[:package],
-                                         repository: @repository, arch: @arch, filename: @filename })
-    end
+    return if @fileinfo # avoid displaying an error for non-existing packages
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/attribute_controller.rb [hunk 3]
```diff
@@ -24,11 +24,9 @@ def new
 
   def edit
     authorize @attribute
-    if @attribute.attrib_type.value_count && ( @attribute.attrib_type.value_count > @attribute.values.length )
-      ( @attribute.attrib_type.value_count - @attribute.values.length ).times do
-        @attribute.values.build(attrib: @attribute)
-      end
-    end
+    return unless @attribute.attrib_type.value_count && ( @attribute.attrib_type.value_count > @attribute.values.length )
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/attribute_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/tag_controller.rb [hunk 4]
```diff
@@ -146,11 +144,9 @@ def get_tags_by_user_and_package( do_render = true )
     @project = @package.project
 
     @tags = @package.tags.where("taggings.user_id = ?", user.id).order(:name)
-    if do_render
-      render partial: "tags"
-    else
-      return @tags
-    end
+    return @tags unless do_render
+
```
Comment (Ana06) [inline src/api/app/controllers/tag_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/tag_controller.rb [hunk 5]
```diff
@@ -130,11 +130,9 @@ def get_tags_by_user_and_project( do_render = true )
     @project = Project.get_by_name(params[:project])
 
     @tags = @project.tags.where("taggings.user_id = ?", user.id).order(:name)
-    if do_render
-      render partial: "tags"
-    else
-      return @tags
-    end
+    return @tags unless do_render
+
```
Comment (Ana06) [inline src/api/app/controllers/tag_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/package_controller.rb [hunk 6]
```diff
@@ -143,12 +143,12 @@ def statistics
...
-      redirect_to controller: 'package', action: :binaries, project: @project,
-                  package: @package, repository: @repository, nextstatus: 404
-      return
-    end
+    return if @statistics
+
+    flash[:error] = "No statistics of a successful build could be found in #{@repository}/#{@arch}"
+    redirect_to controller: 'package', action: :binaries, project: @project,
+                package: @package, repository: @repository, nextstatus: 404
+    return
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: Is this return needed?

File: src/api/app/controllers/webui/package_controller.rb [hunk 7]
```diff
@@ -197,11 +194,11 @@ def binaries
...
-    unless @buildresult
-      flash[:error] = "Package \"#{@package}\" has no build result for repository #{@repository}"
-      redirect_to controller: :package, action: :show, project: @project, package: @package, nextstatus: 404
-      return
-    end
+    return if @buildresult
+
+    flash[:error] = "Package \"#{@package}\" has no build result for repository #{@repository}"
+    redirect_to controller: :package, action: :show, project: @project, package: @package, nextstatus: 404
+    return
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: Is this return needed?

File: src/api/app/controllers/webui/package_controller.rb [hunk 8]
```diff
@@ -351,12 +348,10 @@ def submit_request
...
     @linkinfo = nil
     lt = @package.backend_package.links_to
-    if lt
-      @linkinfo = { package: lt, error: @package.backend_package.error }
-      if lt.backend_package.verifymd5 != @package.backend_package.verifymd5
-        @linkinfo[:diff] = true
-      end
-    end
+    return unless lt
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/package_controller.rb [hunk 9]
```diff
@@ -765,10 +760,8 @@ def view_file
       redirect_back(fallback_location: { action: :show, project: @project, package: @package })
       return
     end
-    if @spider_bot
-      render template: 'webui/package/simple_file_view'
-      return
-    end
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/package_controller.rb [hunk 10]
```diff
@@ -765,10 +760,8 @@ def view_file
...
+    render(template: 'webui/package/simple_file_view') && return if @spider_bot
```
Comment (Ana06) [inline src/api/app/controllers/webui/package_controller.rb]: is the return needed?

File: src/api/app/controllers/webui/patchinfo_controller.rb [hunk 11]
```diff
@@ -91,10 +91,10 @@ def read_patchinfo
     @relogin = @file.has_element?('relogin_needed')
     @reboot = @file.has_element?('reboot_needed')
     @zypp_restart_needed = @file.has_element?('zypp_restart_needed')
-    if @file.has_element?('stopped')
-      @block = true
-      @block_reason = @file.value(:stopped)
-    end
+    return unless @file.has_element?('stopped')
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/patchinfo_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/project_controller.rb [hunk 12]
```diff
@@ -582,11 +582,11 @@ def prjconf
     sliced_params.permit!
 
     @content = @project.config.to_s(sliced_params.to_h)
-    unless @content
-      flash[:error] = @project.config.errors.full_messages.to_sentence
-      redirect_to controller: 'project', nextstatus: 404
-      return
-    end
+    return if @content
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/project_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/project_controller.rb [hunk 13]
```diff
@@ -582,11 +582,11 @@ def prjconf
...
+    flash[:error] = @project.config.errors.full_messages.to_sentence
+    redirect_to controller: 'project', nextstatus: 404
+    return
```
Comment (Ana06) [inline src/api/app/controllers/webui/project_controller.rb]: Is this return needed?

File: src/api/app/controllers/webui/request_controller.rb [hunk 14]
```diff
@@ -142,10 +142,10 @@ def sourcediff
   def require_request
     required_parameters :number
     @req = BsRequest.find_by_number params[:number]
-    unless @req
-      flash[:error] = "Can't find request #{params[:number]}"
-      redirect_back(fallback_location: user_show_path(User.current)) && return
-    end
+    return if @req
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/request_controller.rb]: Why leaving an empty line?

File: src/api/app/controllers/webui/webui_controller.rb [hunk 15]
```diff
@@ -273,11 +273,11 @@ def require_configuration
...
-    if User.current.nil? || !User.current.is_admin?
-      flash[:error] = 'Requires admin privileges'
-      redirect_back(fallback_location: { controller: 'main', action: 'index' })
-      return
-    end
+    return unless User.current.nil? || !User.current.is_admin?
+
+    flash[:error] = 'Requires admin privileges'
+    redirect_back(fallback_location: { controller: 'main', action: 'index' })
+    return
```
Comment (Ana06) [inline src/api/app/controllers/webui/webui_controller.rb]: Is this return needed?

File: src/api/app/controllers/webui/webui_controller.rb [hunk 16]
```diff
@@ -273,11 +273,11 @@ def require_configuration
 
   # Before filter to check if current user is administrator
   def require_admin
-    if User.current.nil? || !User.current.is_admin?
-      flash[:error] = 'Requires admin privileges'
-      redirect_back(fallback_location: { controller: 'main', action: 'index' })
-      return
-    end
+    return unless User.current.nil? || !User.current.is_admin?
+
```
Comment (Ana06) [inline src/api/app/controllers/webui/webui_controller.rb]: Why leaving an empty line?

File: src/api/app/helpers/event_mailer_helper.rb [hunk 17]
```diff
@@ -5,12 +5,10 @@ def project_or_package_text(project, package, opts = {})
...
       text = project
     end
-    if opts[:short].nil?
-      if package.present?
-        "package #{text}"
-      else
-        "project #{text}"
-      end
-    end
+
```
Comment (Ana06) [inline src/api/app/helpers/event_mailer_helper.rb]: Why leaving an empty line?

File: src/api/app/helpers/event_mailer_helper.rb [hunk 18]
```diff
@@ -5,12 +5,10 @@ def project_or_package_text(project, package, opts = {})
...
+    return unless opts[:short].nil?
+    return "package #{text}" if package.present?
+
```
Comment (Ana06) [inline src/api/app/helpers/event_mailer_helper.rb]: Why leaving an empty line?

File: src/api/app/mixins/has_relationships.rb [hunk 19]
```diff
@@ -158,16 +158,11 @@ def find!(id)
...
-          raise SaveError, "unknown group '#{id}' on LDAP server"
-        end
-      else
-        raise SaveError, "unknown group '#{id}'"
-      end
+      raise SaveError, "unknown group '#{id}'" unless CONFIG['ldap_mode'] == :on && CONFIG['ldap_group_support'] == :on
+      raise SaveError, "unknown group '#{id}' on LDAP server" unless UserLdapStrategy.find_group_with_ldap(id)
+
+      logger.debug "Find and Create group '#{id}' from LDAP"
+      return Group.create!(title: id)
```
Comment (Ana06) [inline src/api/app/mixins/has_relationships.rb]: the return is not needed

File: src/api/app/models/branch_package.rb [hunk 20]
```diff
@@ -367,9 +367,9 @@ def expand_rev_to_srcmd5(p)
     dir = Directory.find(project: params[:project], package: params[:package], rev: params[:rev])
     raise InvalidFilelistError.new 'no such revision' unless dir
     p[:rev] = dir.value(:srcmd5)
-    unless p[:rev]
-      raise InvalidFilelistError.new 'no srcmd5 revision found'
-    end
+    return if p[:rev]
+
```
Comment (Ana06) [inline src/api/app/models/branch_package.rb]: Why leaving an empty line?

File: src/api/app/models/branch_package.rb [hunk 21]
```diff
@@ -591,8 +591,8 @@ def set_target_project
       end
       @auto_cleanup = ::Configuration.cleanup_after_days
     end
-    if @target_project && !Project.valid_name?(@target_project)
-      raise InvalidProjectNameError, "invalid project name '#{@target_project}'"
-    end
+    return unless @target_project && !Project.valid_name?(@target_project)
+
```
Comment (Ana06) [inline src/api/app/models/branch_package.rb]: Why leaving an empty line?

File: src/api/app/models/bs_request.rb [hunk 22]
```diff
@@ -103,9 +103,9 @@ def check_creator
     unless user
       errors.add(:creator, "Invalid creator specified #{creator}")
     end
-    unless user.is_active?
-      errors.add(:creator, "Login #{user.login} is not an active user")
-    end
+
```
Comment (Ana06) [inline src/api/app/models/bs_request.rb]: Why leaving an empty line?

File: src/api/app/models/bs_request.rb [hunk 23]
```diff
@@ -403,22 +403,22 @@ def permission_check_addreview!
 
   def permission_check_change_groups!
     # adding and removing of requests is only allowed for groups
-    if bs_request_actions.first.action_type != :group
-      raise GroupRequestSpecial.new "Command is only valid for group requests"
-    end
+    return unless bs_request_actions.first.action_type != :group
+
```
Comment (Ana06) [inline src/api/app/models/bs_request.rb]: Why leaving an empty line?

File: src/api/app/models/bs_request.rb [hunk 24]
```diff
@@ -717,9 +717,9 @@ def change_review_state(new_review_state, opts = {})
   end
 
   def check_if_valid_review!(opts)
-    if !opts[:by_user] && !opts[:by_group] && !opts[:by_project]
-      raise InvalidReview.new
-    end
+    return unless !opts[:by_user] && !opts[:by_group] && !opts[:by_project]
+
```
Comment (Ana06) [inline src/api/app/models/bs_request.rb]: Why leaving an empty line?

File: src/api/app/models/bs_request.rb [hunk 25]
```diff
@@ -800,10 +800,10 @@ def setincident(incident)
       end
     end
 
-    if touched
-      save!
-      HistoryElement::RequestSetIncident.create(p)
-    end
+    return unless touched
+
```
Comment (Ana06) [inline src/api/app/models/bs_request.rb]: Why leaving an empty line?

File: src/api/app/models/bs_request_action_maintenance_release.rb [hunk 26]
```diff
@@ -129,9 +129,9 @@ def check_permissions!
 
     # creating release requests is also locking the source package, therefore we require write access there.
     spkg = Package.find_by_project_and_name source_project, source_package
-    unless spkg || !User.current.can_modify_package?(spkg)
-      raise LackingReleaseMaintainership.new 'Creating a release request action requires maintainership in source package'
-    end
+    return if spkg || !User.current.can_modify_package?(spkg)
+
```
Comment (Ana06) [inline src/api/app/models/bs_request_action_maintenance_release.rb]: Why leaving an empty line?

File: src/api/app/models/buildresult.rb [hunk 27]
```diff
@@ -50,11 +50,9 @@ def self.avail_status_values
 
   def self.code2index(code)
     index = AVAIL_STATUS_VALUES[code.to_sym]
-    if index
-      index
-    else
-      raise ArgumentError, "code '#{code}' unknown #{AVAIL_STATUS_VALUES.inspect}"
-    end
+    return index if index
+
```
Comment (Ana06) [inline src/api/app/models/buildresult.rb]: Why leaving an empty line?

File: src/api/app/models/download_repository.rb [hunk 28]
```diff
@@ -14,10 +14,9 @@ class DownloadRepository < ApplicationRecord
...
   def architecture_inclusion
     # Workaround for rspec validation test (validate_presence_of(:repository_id))
-    if repository
-      unless repository.architectures.pluck(:name).include?(arch)
-        errors.add(:base, "Architecture has to be available via repository association.")
-      end
-    end
+    return unless repository
+    return if repository.architectures.pluck(:name).include?(arch)
+
```
Comment (Ana06) [inline src/api/app/models/download_repository.rb]: Why leaving an empty line?

File: src/api/app/models/issue_tracker.rb [hunk 29]
```diff
@@ -83,12 +83,12 @@ def update_issues_bugzilla
...
-      self.issues_updated = @update_time_stamp
-      save!
+    return unless private_fetch_issues(ids)
 
-      return true
-    end
+    self.issues_updated = @update_time_stamp
+    save!
+
+    return true
```
Comment (Ana06) [inline src/api/app/models/issue_tracker.rb]: `save!` already returns `true` so you can remove this line.

File: src/api/app/models/issue_tracker.rb [hunk 30]
```diff
@@ -347,9 +346,9 @@ def start_element(name, attrs = [])
   end
 
   def characters(content)
-    if @@isDesc
-      @@mySummary += content.chomp
-    end
+    return unless @@isDesc
+
```
Comment (Ana06) [inline src/api/app/models/issue_tracker.rb]: Why leaving an empty line?

File: src/api/app/models/project.rb [hunk 31]
```diff
@@ -455,9 +455,9 @@ def check_write_access!(ignoreLock = nil)
     return if Rails.env.test? && User.current.nil? # for unit tests
 
     # the can_create_check is inconsistent with package class check_write_access! check
-    unless check_write_access(ignoreLock)
-      raise WritePermissionError, "No permission to modify project '#{name}' for user '#{User.current.login}'"
-    end
+    return if check_write_access(ignoreLock)
+
```
Comment (Ana06) [inline src/api/app/models/project.rb]: Why leaving an empty line?

File: src/api/app/models/project.rb [hunk 32]
```diff
@@ -549,17 +548,14 @@ def can_be_unlocked?(with_exception = true)
...
-      if with_exception
-        raise ProjectNotLocked.new "project '#{name}' is not locked"
-      else
-        errors.add(:base, 'is not locked')
-      end
-    end
-    if errors.any?
-      return false
+      raise ProjectNotLocked.new "project '#{name}' is not locked" if with_exception
+
```
Comment (Ana06) [inline src/api/app/models/project.rb]: Why leaving an empty line?

File: src/api/app/models/project.rb [hunk 33]
```diff
@@ -1552,10 +1550,10 @@ def unlock(comment = nil)
 
   def unlock_by_request(request)
     f = flags.find_by_flag_and_status('lock', 'enable')
-    if f
-      flags.delete(f)
-      store(comment: "Request got revoked", request: request, lowprio: 1)
-    end
+    return unless f
+
```
Comment (Ana06) [inline src/api/app/models/project.rb]: Why leaving an empty line?

File: src/api/app/models/project.rb [hunk 34]
```diff
@@ -1687,10 +1685,10 @@ def source_file(file, opts = {})
   # FIXME: will be cleaned up after implementing FATE #308899
   def prepend_kiwi_config
     prjconf = source_file('_config')
-    unless prjconf =~ /^Type:/
-      prjconf = "%if \"%_repository\" == \"images\"\nType: kiwi\nRepotype: none\nPatterntype: none\n%endif\n" << prjconf
-      Suse::Backend.put_source(source_path('_config'), prjconf)
-    end
+    return if prjconf =~ /^Type:/
+
```
Comment (Ana06) [inline src/api/app/models/project.rb]: Why leaving an empty line?

File: src/api/app/models/relationship.rb [hunk 35]
```diff
@@ -81,10 +81,10 @@ def self.add_user(obj, user, role, ignoreLock = nil, check = nil)
 
     logger.debug "adding user: #{user.login}, #{role.title}"
     r = obj.relationships.build(user: user, role: role)
-    if r.invalid?
-      logger.debug "invalid: #{r.errors.inspect}"
-      r.delete
-    end
+    return unless r.invalid?
+
```
Comment (Ana06) [inline src/api/app/models/relationship.rb]: Why leaving an empty line?

File: src/api/app/models/user.rb [hunk 36]
```diff
@@ -1005,9 +1006,9 @@ def can_modify_project_internal(project, ignoreLock)
   # Hashes the given parameter by the selected hashing method. It uses the
   # "password_salt" property's value to make the hashing more secure.
   def hash_string(value)
-    if password_hash_type == "md5"
-      Digest::MD5.hexdigest(value + password_salt)
-    end
+    return unless password_hash_type == "md5"
+
```
Comment (Ana06) [inline src/api/app/models/user.rb]: Why leaving an empty line?

File: src/api/test/functional/webui/users_test.rb [hunk 37]
```diff
@@ -98,10 +98,10 @@ def test_add_and_edit_project_users # spec/support/shared_examples/features/user
   # Test Helpers
 
   def edit_role(cell, new_value)
-    unless new_value.nil?
-      input = cell.first(:css, 'input')
-      input.click unless input.selected? == new_value
-    end
+    return if new_value.nil?
+
```
Comment (Ana06) [inline src/api/test/functional/webui/users_test.rb]: Why leaving an empty line?

**Comment by Ana06** [comment]
Also, why are there two commits? :confused:

**Comment by evanrolfe** [comment]
@Ana06 there are two commits because david start this PR and I took over. So the first commit is david's changes and the second is my changes.

File: src/api/app/controllers/tag_controller.rb (hunk 4 above)
Comment (evanrolfe) [inline src/api/app/controllers/tag_controller.rb]: because i felt like it? lol, i think it looks nicer but its not a big deal so i will get rid of them..

Comment (DavidKang) [inline src/api/app/controllers/tag_controller.rb]: I'm with @evanrolfe.

Comment (evanrolfe) [inline src/api/app/controllers/tag_controller.rb]: @Ana06 should i delete the empty lines I've added or can I leave them?

Comment (Ana06) [inline src/api/app/controllers/tag_controller.rb]: According to [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#empty-lines-between-methods):

> Use empty lines between method definitions and also to break up methods into logical paragraphs internally.

And this is not only something for Ruby, as it is something standard in most programming languages to use blank lines inside methods only to separate logical sections, for example you can find it in the [Python](https://www.python.org/dev/peps/pep-0008/#blank-lines) and [Java](http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html) style guides.

And I would say than in most of the cases your empty lines are not separating logical sections and consequently they shouldn't be there. When there is a lot of code after the return I could agree that are different logical sections and that this improve readability, but when there is only one more line I really don't see the point.

**Comment by Ana06** [comment]
@evanrolfe 
> @Ana06 there are two commits because david start this PR and I took over. So the first commit is david's changes and the second is my changes.

If they are the same thing you could squash them (you can add both name in the commit message and github already show the two pictures) and if not change the commit message because it is quite confusing having two different commits with the same message

**Comment by evanrolfe** [comment]
> If they are the same thing you could squash them (you can add both name in the commit message and github already show the two pictures) and if not change the commit message because it is quite confusing having two different commits with the same message

@Ana06 IMO I prefer not to squash them because there are quite a lot of changes in both commits and we should be able to see who has commited what changes, i.e. if you go on git blame you should be able to see which person is responsible for the changes, if I squash them we lose that knowledge..

File: src/api/app/controllers/tag_controller.rb (hunk 4 above)
Comment (evanrolfe) [inline src/api/app/controllers/tag_controller.rb]: ok i'll delete the blank lines

**Comment by Ana06** [comment]
[Enable the Style/GuardClause cop in Rubocop](https://trello.com/c/ZXHXwSOM/178-enable-the-style-guardclause-cop-in-rubocop)

**Comment by evanrolfe** [comment]
@Ana06 I've deleted the empty lines and implemented your suggestions. Please re-review

**Pull request merged**